### PR TITLE
Fix housecolor crash and improve readability of relevant code in GameAssetManager

### PR DESCRIPTION
**Pull request by xezon**
Loading **Tournament Desert** map crashes the game.

### CODE REVIEW

**Review by duncanspumpkin — CHANGES_REQUESTED** [review]
Can you just change the nullptr check to the correct way round it would be much cleaner.
`const char *str2 = str != nullptr ? str + 1 : mesh->Get_Name();`

File: src/w3d/renderer/assetmgr.cpp [hunk 1]
```diff
@@ -758,7 +758,7 @@ bool Get_Mesh_Color_Methods(MeshClass *mesh, bool *housecolor, bool *zhc)
     if (matinfo != nullptr) {
         for (int i = 0; i < matinfo->Texture_Count(); i++) {
 
-            if (!strncasecmp(matinfo->Peek_Texture(i)->Get_Name(), "ZHC", 3)) {
+            if (strncasecmp(matinfo->Peek_Texture(i)->Get_Name(), "ZHC", 3) == 0) {
```
Comment (duncanspumpkin) [inline src/w3d/renderer/assetmgr.cpp]: unrelated revert

File: src/w3d/renderer/assetmgr.cpp [hunk 2]
```diff
@@ -767,11 +767,12 @@ bool Get_Mesh_Color_Methods(MeshClass *mesh, bool *housecolor, bool *zhc)
...
-    const char *str = strchr(mesh->Get_Name(), '.');
-    const char *str2 = str == nullptr ? str + 1 : mesh->Get_Name();
+    captainslog_assert(mesh->Get_Name() != nullptr);
+    const char *word = strchr(mesh->Get_Name(), '.');
+    word = (word != nullptr) ? word + 1 : mesh->Get_Name();
 
-    if (str2 != nullptr) {
-        if (!strncasecmp(str2, "HOUSECOLOR", 10)) {
+    if (word != nullptr) {
+        if (strncasecmp(word, "HOUSECOLOR", 10) == 0) {
```
Comment (duncanspumpkin) [inline src/w3d/renderer/assetmgr.cpp]: unrelated revert

File: src/w3d/renderer/assetmgr.cpp [hunk 3]
```diff
@@ -866,7 +869,7 @@ bool GameAssetManager::Recolor_Mesh(RenderObjClass *robj, uint32_t colour)
 
     for (int i = 0; i < matinfo->Texture_Count(); i++) {
         TextureClass *texture = matinfo->Peek_Texture(i);
-        if (!strncasecmp(texture->Get_Name(), "ZHC", 3)) {
+        if (strncasecmp(texture->Get_Name(), "ZHC", 3) == 0) {
```
Comment (duncanspumpkin) [inline src/w3d/renderer/assetmgr.cpp]: revert

**Comment by xezon** [comment]
I updated the title of the change. That should do (Squash and merge). No need to revert it all.

File: src/w3d/renderer/assetmgr.cpp (hunk 3 above)
Comment (xezon) [reply src/w3d/renderer/assetmgr.cpp]: fixed.

**Review by duncanspumpkin — APPROVED** [review]
I personally dont see this as more readable but its fine.

**Comment by xezon** [comment]
On second look, it looks as if the `if (word != nullptr)` branch could be removed, no? At this point word is not to be expected null.

**Comment by duncanspumpkin** [comment]
It can only be nullptr if get name is nullptr. I don't really want to think through all possibilities

**Comment by xezon** [comment]
If `Get_Name()` was null, then it would crash 2 lines before already at `strchr(mesh->Get_Name(), '.');`. That's why I placed the assert just before that.

**Comment by duncanspumpkin** [comment]
I don't think it is guaranteed to crash in that function standard doesn't specify if strchr is nullptr safe.

**Comment by xezon** [comment]
I check later. If `strchr` was checking null, then the `assert` would be obsolete. If `strchr` is not checking null, then the assert is ok and the condition is obsolete.

File: src/w3d/renderer/assetmgr.cpp (hunk 1 above)
Comment (xezon) [reply src/w3d/renderer/assetmgr.cpp]: reverted

**Pull request merged**